Code/Mantid/Framework/PythonInterface/plugins/algorithms/GetNuMegMuonicXRD.py

```diff
@@ -0,0 +1,63 @@
+from mantid.api import * # PythonAlgorithm, registerAlgorithm, WorkspaceProperty
+from mantid.kernel import *
+
+#pylint: disable=no-init
+class GetNegMuMuonicXRD(PythonAlgorithm):
+    #Dictionary of <element>:<peaks> easily extendible by user.
+    Muonic_XR={'Au' :[8135.2,8090.6,8105.4,8069.4,5764.89,5594.97,3360.2,
+                       3206.8,2474.22,2341.21,2304.44,1436.05,1391.58,1104.9,
+                       899.14,869.98,405.654,400.143],
+               'Ag': [3184.7,3147.7,901.2,869.2,308.428,304.759],
+               'Cu' :[1512.78,1506.61,334.8,330.26],
+               'Zn' :[1600.15,1592.97,360.75,354.29],
+               'Pb' :[8523.3,8442.11,5966.0,5780.1,2641.8,2499.7,
+                        2459.7,1511.63,1214.12,1028.83,972.3,938.4,
+                        437.687,431.285],
+               'As' : [1866.9,1855.8,436.6,427.5],
+               'Sn' : [3457.3,3412.8,1022.6,982.5,349.953,345.226]}
+
+    def PyInit(self):
+        element_type = self.Muonic_XR.keys()
+        self.declareProperty("ElementList", "None", StringListValidator(element_type),
+                             doc="List of available elements")
+        self.declareProperty(StringArrayProperty("Elements", values=[],
+                             direction=Direction.Input
+                             ))
+        self.declareProperty(name="YAxisPosition",
+                                    defaultValue=-0.001,
+                                    doc="Position for Markers on the y-axis")
+
+    def get_Muonic_XR(self, element):
```

Comment:
About the names of methods and functions, in principle we follow the more Pythonic way: lower case and _ instead of the CamelCase convention that we use in C++ code. So you would use `get_muonix_xr` (well the XR could be uppercase), create_`muonic_xr_ws`, etc. See http://www.mantidproject.org/How_to_fix_Pylint_Warnings#Naming. Now in practice I'd say that much if not a majority of out Python code is CamelCase. And of course the PyInit and PyExec methods have a fixed name.

Also if you can use a single `_` as prefix in the name of methods to indicate that they are intended to be private.